### PR TITLE
Bw 609

### PR DESCRIPTION
**Pull request by milansar**

### CODE REVIEW

File: projects/shared-library/src/lib/core/auth/web/firebase-auth.service.ts [hunk 1]
```diff
@@ -88,4 +93,28 @@ export class WebFirebaseAuthService implements FirebaseAuthService {
     public resumeState(user) {
 
     }
+
+    public updatePushToken(token: string) {
+        this.pushToken = token;
+    }
+
+    public updateOnConnect(user: User) {
+        this.db.object(`${CollectionConstants.INFO}/${CollectionConstants.CONNECTED}`)
```
Comment (milansar) [inline projects/shared-library/src/lib/core/auth/web/firebase-auth.service.ts]: add comment that this is inbuilt document and will not be visible in firebase console

File: projects/shared-library/src/lib/environments/bitwiser-edu/environment.prod.ts [hunk 2]
```diff
@@ -15,10 +15,12 @@ export const CONFIG: IConfig = {
     messagingSenderId: '704157888306'
   },
   'functionsUrl': `https://bitwiser-edu.firebaseapp.com/${appConstants.API_VERSION}`,
-  'editorUrl': 'https://bitwiser-edu.firebaseapp.com/trivia-editor',
+  'editorUrl': 'https://bitwiser-edu.firebaseapp.com/editor',
```
Comment (milansar) [inline projects/shared-library/src/lib/environments/bitwiser-edu/environment.prod.ts]: are you sure editor path is correct, as per angula.json it still output editor into trivia-editor path

File: projects/shared-library/src/lib/environments/bitwiser-edu/environment.prod.ts [hunk 3]
```diff
@@ -15,10 +15,12 @@ export const CONFIG: IConfig = {
...
   'hightlighJsURL' : 'https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js',
   'hightlighCSSURL' : 'https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/a11y-light.min.css',
   'katexCSSURL' : 'https://cdn.jsdelivr.net/npm/katex@0.10.1/dist/katex.min.css',
+  'addressByLatLongURL': 'https://maps.googleapis.com/maps/api/geocode/json',
```
Comment (milansar) [inline projects/shared-library/src/lib/environments/bitwiser-edu/environment.prod.ts]: This needs to be removed as it is read from constant file

File: hooks/before-prepare/nativescript-copy-configuration-files.js [hunk 4]
```diff
@@ -1,47 +0,0 @@
-
-var nativeScriptConfig = require("./../../custom-hooks/nativescript-copy-configuration-files");
```
Comment (milansar) [inline hooks/before-prepare/nativescript-copy-configuration-files.js]: why this file was deleted?

File: hooks/before-watch/nativescript-copy-configuration-files.js [hunk 5]
```diff
@@ -1,46 +0,0 @@
-
-var nativeScriptConfig = require("./../../custom-hooks/nativescript-copy-configuration-files");
```
Comment (milansar) [inline hooks/before-watch/nativescript-copy-configuration-files.js]: why this file was deleted?

File: firebase-rules.json [hunk 6]
```diff
@@ -1,46 +1,12 @@
...
-        													    data.child('status').val() != 2)
-                                          ||
-                                      (!data.exists() && 
-                                       newData.child('created_uid').val() == auth.uid))"
-      }
+    ".read": "auth != null",
+    ".write": "auth != null",
+    "users":{
+      "$token_id":{      
+          ".read": "auth.uid!== null",
```
Comment (milansar) [inline firebase-rules.json]: user should be able to read its own data

**Comment by milansar** [comment]
@anihalaney , PR is ready for your review

File: firebase-rules.json [hunk 7]
```diff
@@ -1,46 +1,12 @@
...
-                                      (!data.exists() && 
-                                       newData.child('created_uid').val() == auth.uid))"
-      }
+    ".read": "auth !== null",
+    ".write": "auth !== null",
+    "users":{
+      "$token_id":{      
+         ".read": "auth.uid !== null && data.child('userId').val() === auth.uid",
+         ".write": "newData.child('userId').val() === auth.uid"      
+        }
```
Comment (anihalaney) [inline firebase-rules.json]: are the rules ok? we are removing a bunch of it

Comment (milansar) [reply firebase-rules.json]: yes, these are rules for realtimeDB which we are no longer using, now only use cases is supporting user online status

File: projects/shared-library/src/lib/shared/mobile/component/drawer-component/drawer-component.ts [hunk 8]
```diff
@@ -180,13 +217,22 @@ export class DrawerComponent implements OnInit, OnDestroy {
     }
 
     resetValues() {
-        this.logOut = false;
-        this.pushToken = undefined;
+
         this.authProvider.logout();
+        /* We have used Timeout because authprovide.logout() nullify object after some time */
+        setTimeout(() => {
```
Comment (anihalaney) [inline projects/shared-library/src/lib/shared/mobile/component/drawer-component/drawer-component.ts]: shouldn't we use windowRef for setTimeout ?

File: projects/trivia/src/app/components/app/app.component.ts [hunk 9]
```diff
@@ -56,8 +58,23 @@ export class AppComponent implements OnInit, OnDestroy {
           this.router.navigate([url]);
         }
 
+        this.intervalSubscription = interval(1000 * 60 * 1)
```
Comment (anihalaney) [inline projects/trivia/src/app/components/app/app.component.ts]: we should also unsubscribe on ngOnDestroy?

File: projects/shared-library/src/lib/shared/mobile/component/drawer-component/drawer-component.ts (hunk 8 above)
Comment (yugank1991) [reply projects/shared-library/src/lib/shared/mobile/component/drawer-component/drawer-component.ts]: it is native script component. so ,we can't use windowRef in it

File: projects/trivia/src/app/components/app/app.component.ts (hunk 9 above)
Comment (yugank1991) [reply projects/trivia/src/app/components/app/app.component.ts]: this.intervalSubscription is already added in subscriptions array which is called on ngdestroy

**Pull request merged**